shell/platform/tizen/key_event_handler.cc

```diff
@@ -30,24 +38,34 @@ Eina_Bool KeyEventHandler::OnKey(void* data, int type, void* event) {
   auto* engine = self->engine_;
   auto is_down = type == ECORE_EVENT_KEY_DOWN;
 
-  if (strcmp(key->keyname, kPlatformBackButtonName) == 0) {
-    // The device back button was pressed.
-    if (engine->navigation_channel && !is_down) {
-      engine->navigation_channel->PopRoute();
-    }
-  } else {
-    if (engine->text_input_channel) {
-      if (is_down) {
-        engine->text_input_channel->OnKeyDown(key);
-      }
-      if (engine->text_input_channel->IsSoftwareKeyboardShowing()) {
-        return ECORE_CALLBACK_PASS_ON;
-      }
+  FT_LOGI("Keycode: %d, name: %s, mods: %d, is_down: %d", key->keycode,
+          key->keyname, key->modifiers, is_down);
+
+  if (engine->text_input_channel) {
+    if (is_down) {
+      engine->text_input_channel->OnKeyDown(key);
     }
-    if (engine->key_event_channel) {
-      engine->key_event_channel->SendKeyEvent(key, is_down);
+    if (engine->text_input_channel->IsSoftwareKeyboardShowing()) {
+      return ECORE_CALLBACK_PASS_ON;
     }
   }
+
+  if (engine->key_event_channel) {
+    auto keyname = std::string(key->keyname);
+    engine->key_event_channel->SendKeyEvent(
+        key, is_down, [engine, keyname, is_down](bool handled) {
+          if (handled || is_down) {
+            return;
+          }
+          if (keyname == kBackKey) {
+            if (engine->navigation_channel) {
+              engine->navigation_channel->PopRoute();
+            }
+          } else if (keyname == kExitKey) {
+            ui_app_exit();
+          }
```

Comment:
This was added due to the second issue in https://github.com/flutter-tizen/flutter-tizen/issues/136. I couldn't test it because I wasn't able to reproduce the issue.

Reply:
I'm just curious and asking.. As I remember.. you think we don't need these handing? 
Why did you change your mind?

Reply:
@bwikbs What made you think I changed my mind? No, I haven't. This changes just addresses the second issue in https://github.com/flutter-tizen/flutter-tizen/issues/136.

Reply:
@swift-kim :ok_hand: